e2e/helpers/hooks_helper.js

```diff
@@ -22,6 +22,11 @@ module.exports = class HooksHelpers extends Helper {
     });
   }
 
+  _beforeStep() {
+    const helper = this.helpers['Puppeteer'] || this.helpers['WebDriver'];
+    return helper.waitForInvisible('xuilib-loading-spinner', 20);
```

Comment:
Does it need to be attached to the return?

Reply:
yes, it returns promise, and framework that executes _beforeStep must get promise, otherwise it will be waiting in background and wont help with anything

Reply:
ahh missed that sentence in the helpers readme https://codecept.io/helpers/#hooks